Nextflow conversion

hyunhwan-bcm opened the pull request:
The PR includes numerous changes and should be carefully reviewed. @sasidharPasupuleti and @ZaahidShaik, please try to run this PR and verify its functionality. It would be beneficial to test it against a previously completed run.

## The PR Summary
- [x] convert the pipeline to a nextflow version
- [x] add error handling strategy 
- [ ] output test 
  - [x] on demo data (demo.vcf.gz, 8.7M)
  - [x] on real data (6656513962b9002680181ed0.recode.vcf.gz, 930K)</strike>
- [ ] integrity with README
- [ ] code review
- [ ] code optimization 

## How to use (or test)

Here is  an example:

```
nextflow run main.nf --run_id 12345 --ref_ver "hg19" --outdir testout_061024 \
                                    --input_hpo /home/hwan/AI_MARRVEL/test_data/hpos.txt \
                                    --input_vcf /home/hwan/AI_MARRVEL/test_data/demo.vcf.gz \
                                    --ref_dir /home/hwan/aim_data_dependencies_04_24_2024_sasi
```

You can also change the parameters in `nextflow.config` - you can run `nextflow run main.nf`

hyunhwan-bcm committed:
ADD INDEX_VCF process

hyunhwan-bcm committed:
added pre-processing process

hyunhwan-bcm committed:
change dirname run -> script.

hyunhwan-bcm committed:
added ANNOT PHRANK/ENSMBL

hyunhwan-bcm committed:
enable docker.

hyunhwan-bcm committed:
add phrank scoring.

hyunhwan-bcm committed:
arrange newlines.

hyunhwan-bcm committed:
add OMIM simmiarity calculation.

hyunhwan-bcm committed:
add OMIM and HPO refs.

hyunhwan-bcm committed:
change for file paths.

hyunhwan-bcm committed:
add output paths.

hyunhwan-bcm committed:
forgot to add genemap2 path.

hyunhwan-bcm committed:
add PHRANK SCORING and HPO SIM calc.

hyunhwan-bcm committed:
add GNOMAD references

hyunhwan-bcm committed:
add   FILTER_PROBAND

hyunhwan-bcm committed:
add VEP_ANNOTATE

hyunhwan-bcm committed:
not sure why is is not working.

hyunhwan-bcm committed:
it was a problem for parameters.

hyunhwan-bcm committed:
let's wait until I get something.

hyunhwan-bcm committed:
fix a typo

hyunhwan-bcm committed:
fixed another typo

hyunhwan-bcm committed:
found the bug, but haven't compared the output and the other parts.

hyunhwan-bcm committed:
change the cache mode to lenient.

hyunhwan-bcm committed:
change all wildcard to params.run_id

hyunhwan-bcm committed:
added prediction, need to validate.

hyunhwan-bcm committed:
replace phrank scripts to bin.

hyunhwan-bcm committed:
all scripts have been moved to bin

hyunhwan-bcm committed:
all scripts have been moved to bin

hyunhwan-bcm committed:
merge

hyunhwan-bcm committed:
remove __init__.py - empty

hyunhwan-bcm committed:
lint the python files.

hyunhwan-bcm committed:
lint the R files.

hyunhwan-bcm committed:
stop tracking .gitignore.

hyunhwan-bcm committed:
update .gitignore.

hyunhwan-bcm committed:
update .gitignore.

hyunhwan-bcm committed:
edit to preserve the selected variants.

hyunhwan-bcm committed:
add TODO

hyunhwan-bcm committed:
change output names to have run_id.

hyunhwan-bcm committed:
change nextflow.config

hyunhwan-bcm committed:
add run_diff to compare outputs.

hyunhwan-bcm committed:
fix the order.

hyunhwan-bcm committed:
clean-up

hyunhwan-bcm committed:
add errorStrategy.

hyunhwan-bcm committed:
fix paraemter typos.

hyunhwan-bcm committed:
clean-up.

hyunhwan-bcm committed:
change the process setup for the Feature Engineering Part 1.

hyunhwan-bcm committed:
change cpu configure for some processes

hyunhwan-bcm committed:
Update main.nf

hyunhwan-bcm committed:
Merge branch 'nextflow_conversion' of github.com:hyunhwan-bcm/AI_MARRVEL into nextflow_conversion

jylee-bcm committed:
Handle MT values as -1 to avoid string comparison (#1)

This change ensures that MT (empty) values are
treated consistently as -1,
improving the speed by eliminating string comparisons.

jylee-bcm committed:
Refactor prediction step to eliminate multiprocessing (#2)

Previously, multiprocessing was utilized in the prediction step, resulting in increased memory consumption proportional
to the number of worker processes.
By refactoring the code to replace multiprocessing with a simple for loop, memory usage has been significantly reduced.
This change not only enhances efficiency by eliminating the overhead of virtual memory but also accelerates runtime performance.

hyunhwan-bcm committed:
Merge branch 'nextflow_conversion' of github.com:hyunhwan-bcm/AI_MARRVEL into nextflow_conversion

jylee-bcm committed:
Optimize duplication checking to index-based comparison (#3)

hyunhwan-bcm committed:
Merge branch 'nextflow_conversion' of github.com:hyunhwan-bcm/AI_MARRVEL into nextflow_conversion

jylee-bcm committed:
Optimize process_sample() Function Using Pandas DataFrame (#5)

* Optimize process_sample() Function Using Pandas DataFrame

The previous implementation of process_sample()
relied on Python built-in for-loops, which were
less efficient. This update replaces the for-loops
with optimized Pandas DataFrame operations
to enhance performance.

* Remove unused old codes related to process_sample()

* Remove an unnecessary line

jylee-bcm committed:
Removed Multiprocessing from feature.py (#6)

The use of multiprocessing in feature.py caused high memory consumption due to the forking mechanism, making it a bottleneck in the project. To maintain performance, indexing techniques have been applied to optimize query speed as a trade-off.

hyunhwan-bcm committed:
Merge branch 'nextflow_conversion' of github.com:hyunhwan-bcm/AI_MARRVEL into nextflow_conversion

hyunhwan-bcm committed:
Fix vep split (#7)

* change the process setup for the Feature Engineering Part 1.

* change cpu configure for some processes

* split was not working by symlink, and it worked with readlink -f

jylee-bcm commented on scripts/proc.sh:
If it's not used anymore, let's remove it

jylee-bcm commented on nextflow.config:
* Rename `nextflow.config` to `nextflow.config.example`
* Add `nextflow.config` to `.gitignore` list
* `nextflow.config.example `contain an instruction saying: Use this example file to create nextflow.config and change the values for: ref_dir, input_vcf, input_hpo`

jylee-bcm commented on .gitignore:
Add nextflow.config

jylee-bcm reviewed (changes requested):
I think `nextflow.config` should contain different values for each environment, so I'd like to include it into `.gitignore`. For the details, please read the comment on related code.

jylee-bcm committed:
Stop generating unused intermediate file (#10)

r1_scores.txt was unused but unnecessarily generated and adds the complexity on the overall code.

jylee-bcm committed:
Optimize Feature Engineering Part 2 (#9)

* Applied scalar multiplication to vector
* Used indexing technique
* Transformed from float64 to float32

jylee-bcm committed:
Fix a bug that vep splitter didn't work under nextflow env (#8)

* Fix a bug that vep splitter didn't work under nextflow env

* Revert "Fix vep split (#7)"

This reverts commit 867c884ec4199a683182ac4a6dc40616cb8e45ce.

jylee-bcm committed:
Untrack Nextflow Config (#11)

* Untrack Nextflow Config

* Keep tracking nextflow config and Use params.yaml

* Update README.md

* Update README.md

hyunhwan-bcm committed:
Update VarTierDiseaseDBFalse.R To Improve Speed

- replace the file reader
- Optimization by unnest gene/variant level operation

hyunhwan-bcm commented:
@jylee-bcm please review my latest commit to this branch.

hyunhwan-bcm committed:
remove a code-line for debug

a value was overridden for testing.

hyunhwan-bcm committed:
Merge branch 'nextflow_conversion' of github.com:hyunhwan-bcm/AI_MARRVEL into nextflow_conversion

jylee-bcm reviewed (commented):
I think it's better to remove /run/ directory, so that git can track the difference.
I will upload PR for it.

hyunhwan-bcm commented:
I thought I remove them, but it wasn't pushed. I am fine that if you delete the directory.

jylee-bcm commented on utils/run_diff.py:
Is it actively used? if not, remove it

jylee-bcm commented:
> I thought I remove them, but it wasn't pushed. I am fine that if you delete the directory.
You were right, just git couldn't track the difference.

hyunhwan-bcm commented:
You can remove it just for comparison of every column of dataframes

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Jaeyeon Lee ***@***.***>
Sent: Friday, July 12, 2024 11:46:43 AM
To: LiuzLab/AI_MARRVEL ***@***.***>
Cc: Jeong, Hyun-Hwan ***@***.***>; Author ***@***.***>
Subject: Re: [LiuzLab/AI_MARRVEL] Nextflow conversion (PR #18)

***CAUTION:*** This email is not from a BCM Source. Only click links or open attachments you know are safe.
________________________________

@jylee-bcm commented on this pull request.

________________________________

On utils/run_diff.py<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_LiuzLab_AI-5FMARRVEL_pull_18-23discussion-5Fr1676191866&d=DwMFaQ&c=ZQs-KZ8oxEw0p81sqgiaRA&r=-lih8OIo66XtgJH6zqJrfjIwRtqkkhDYbiD5mmg9N6M&m=2XM5e49yxOQhs1NNB4SFoGxj8unTaRLRVGntuz7wnTDY0WvZgWrEuhHM-xf7btMe&s=CdcZ_jtWIDD_2TTxmQ-_uEx2wfQOROk4kegA_jx-4T4&e=>:

Is it actively used? if not, remove it

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_LiuzLab_AI-5FMARRVEL_pull_18-23pullrequestreview-2D2175324846&d=DwMFaQ&c=ZQs-KZ8oxEw0p81sqgiaRA&r=-lih8OIo66XtgJH6zqJrfjIwRtqkkhDYbiD5mmg9N6M&m=2XM5e49yxOQhs1NNB4SFoGxj8unTaRLRVGntuz7wnTDY0WvZgWrEuhHM-xf7btMe&s=bBMfWwWhnMxRg3npFSHUehyCkqgCx2jq4VKJsAVZqh8&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_BHZAYJCDHCVNBEOBD4ZZX73ZMAB7HAVCNFSM6AAAAABJCZNYZSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCNZVGMZDIOBUGY&d=DwMFaQ&c=ZQs-KZ8oxEw0p81sqgiaRA&r=-lih8OIo66XtgJH6zqJrfjIwRtqkkhDYbiD5mmg9N6M&m=2XM5e49yxOQhs1NNB4SFoGxj8unTaRLRVGntuz7wnTDY0WvZgWrEuhHM-xf7btMe&s=NnDYkt65T14WXrysptKP-JYiLilodoZWFnLhVKKvRvc&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

jylee-bcm committed:
Remove unused rundiff.py (#12)

jylee-bcm reviewed (approved):
LGTM

hyunhwan-bcm committed:
Change library in VarTierDiseaseDBFalse.R (#13)

* remove scripts/

* change imported library; tidyverse is not working - why?

hyunhwan-bcm committed:
remove commented lines. (#14)

* rebase commit https://github.com/LiuzLab/AI_MARRVEL/commit/16b72dde8d1f72fc4949b2b3919c72c8e2342c5a

* docs: Update installation.rst

* edit dockerfile for automatical bedtools/bcftools installation.

* remove bcftools/bedtools installation.

* fix Dockerfile for R installation.

* ci: add GitHub Actions for automated testing

* paper link updated

* remove `dplyr`

`tidyverse` has `dplyr`, so no need to install twice.

* remove commented lines.

---------

Co-authored-by: ZaahidShaik <zahidshaik01@gmail.com>
Co-authored-by: SasidharPasupuleti <34906695+sasidharPasupuleti@users.noreply.github.com>
Co-authored-by: Sun-Young Kim <seonyouk@gmail.com>
Co-authored-by: ChaozhongLiu <czliubioinfo@gmail.com>

jylee-bcm committed:
Filter VCFs to Exonic Regions Only (#17)

To prevent it from running more than days with the input vcf of WGS (Whole Genome Sequence),
this changes filter to exonic regions only, based on the BED file generated with UCSC genome browser, by merging all exome probeset tracks (union function) (Mohammed provided)

This change ensures the pipeline can process within 30 minutes with the user's input of WGS.

jylee-bcm committed:
Preprocess for normalizing mod5 network (#16)

* Separate process for normalizing mod5 network

* Adjust memory and cpus setting for nextflow processes

* Preprocess Mod5 Diffusion

* Remove an empty line

jylee-bcm committed:
Revert "Filter VCFs to Exonic Regions Only (#17)" (#18)

This reverts commit 716bcd5c69b229528ee2a2e817c3a0b2a86ec312.

jylee-bcm committed:
Revert "Revert "Filter VCFs to Exonic Regions Only (#17)" (#18)" (#20)

This reverts commit 5f19be6baee934b04eb21ae28e50c2b8f4d9b0b6.

jylee-bcm committed:
Reorder exonic filtering step to after mitochondria removal (#21)

* Quick Fix

* Reorder

jylee-bcm committed:
Update bcftools to 1.20 (#22)

hyunhwan-bcm committed:
Hj/nxf comp/update bivar (#27)

* revert some files to main.

* fix the known sorting issue.

* make process_gene same as main_fix_sort branch

* conf4 are now the same.

* force var_grp to be sorted by the index.

* add indexing sorting for the var index.

* temporal off of the dedup.

* change the sorting for index of each group df.

* make confidence sorting stable.

hyunhwan-bcm committed:
Check completed - nextflow comparison with main using hg38 (#25)

* revert some files to main.

* fix the known sorting issue.

* make process_gene same as main_fix_sort branch

* conf4 are now the same.

* force var_grp to be sorted by the index.

* fetching bin/generate_new_matrix_2.py from nextflow_coversion, checked output. (#26)

* Revert 27 hj/nxf comp/update bivar (#28)

* Hj/nxf comp/update bivar (#27)

* revert some files to main.

* fix the known sorting issue.

* make process_gene same as main_fix_sort branch

* conf4 are now the same.

* force var_grp to be sorted by the index.

* add indexing sorting for the var index.

* temporal off of the dedup.

* change the sorting for index of each group df.

* make confidence sorting stable.

* Revert "Hj/nxf comp/update bivar (#27)"

This reverts commit 95af06491f12b25b83692edf955c767beb9e30a0.

* pulling nextflow_conversion branch files.

hyunhwan-bcm committed:
Merge branch 'hyunhwan-bcm-nextflow_conversion' into nextflow_conversion

hyunhwan-bcm committed:
checkout confict files from main
